packages/neon-core/src/tx/components/Cosigner.ts

```diff
@@ -0,0 +1,118 @@
+import { WitnessScope } from "./WitnessScope";
+import { StringStream, num2hexstring } from "../../u";
+import { deserializeArrayOf, serializeArrayOf } from "../main";
+
+export interface CosignerLike {
+  account: string;
+  scopes: number;
+  allowedContracts: string[];
+  allowedGroups: string[];
+}
+
+export class Cosigner {
+  /**
+   * script hash of cosigner
+   */
+  public account: string;
+  public scopes: WitnessScope;
+  public allowedContracts: string[];
+  public allowedGroups: string[];
+
+  /**
+   * @description This limits maximum number of allowedContracts or allowedGroups here
+   */
+  private readonly MAX_SUB_ITEMS: number = 16;
+
+  public constructor(signer: Partial<CosignerLike>) {
+    const { account, scopes, allowedContracts, allowedGroups } = signer;
+    this.account = account || "";
+    this.scopes = !scopes ? WitnessScope.Global : scopes & 0x11;
+    this.allowedContracts = allowedContracts || [];
+    this.allowedGroups = allowedGroups || [];
+  }
+
+  public get isGlobal() {
```

Comment:
why have the getters when the user has clear access to the scopes. Let the user do the comparison we dont need such methods.